blog/content/post/2021-02-21-tigerlake-osf.md

```diff
@@ -0,0 +1,183 @@
+error. The typical different between LPDDR and DDR memory is that the former
+requires an exact mapping of memory signal from CPU to DRAM to be passed to
+FSP. So simply zeroing this configuration should be enough. Also the memory
+initialization was called by `meminit_lpddr4x` which was not suited for DDR4,
+so I have to change it to `meminit_ddr4` with appropriate parameters for SMBus
+SPD addresses. After these modifications I could past through the memory
+initialization.
+
+Another problem faced was the CPU initialization in coreboot. I was getting
+board resets during MCE (Machine Check Error registers) clearing which I have
+written about here:
+
+https://mail.coreboot.org/hyperkitty/list/coreboot@coreboot.org/thread/MW44TIEMFMVDWPVCAFPE2QUFXXGZYYAX/
+
+Till today I haven't got any response, however, I managed to resolve the
+problem as well. When investigating the logs I noticed that the microcode is
+not automatically loaded before the reset vector. For reminder, the microcode
+is being loaded through FIT table before reset vector since Haswell (4th
+generation) processors. This could be due to the processor being engineering
+sample (which is common for RVP platforms). I had placed explicit call to
+update microcode using coreboot then I got past the PCI enumeration phase.
+
+Right after the PCI enumeration phase I was hit with FSP notify error. No idea
+what could cause this issue, since the notify phases typically do not do much,
+but yet I managed to hit an error. To this point I haven't been able to figure
+out what is wrong. Trying to narrow it down with debug FSP binary didn't help
+as well, because the FSP asserts in Thunderbolt/USB type C initialization.
+
+I am still not close to booting an operating system, but yet I have encountered
+many bugs in the meantime. This is the first time I have been put through so
+much effort to boot a reference platform. Previous platform I have tried
+(Cometlake-S or Kaby Lake U) did not cause almost any problem. The quality of
+the implementation of the Intel silicon support done by Intel employees
+themselves seem to have degraded. As a proof of that please red the following
+section about EDK2 MinPlatform.
+
+## EDK2
+
+EDK2 besides the open source common modules for UEFI compliant firmware it also
+contains the platform code hosted in a separate repository called
+[edk2-platforms](https://github.com/tianocore/edk2-platforms). This is where
+Intel publishes the reference board support code that integrates with FSP to
+boot reference platforms. The TigerLake open board packages have been published
+just few days ago. I gave it a try almost immediately with hope it will give
+better results.
+
+But on high hopes it ended, quite quickly. Although EDK2 is supposed to support
+GCC5, it occurred it is not always true. The freshly published code was not
+buildable with GCC5 when trying with the 3mdeb
+[edk2-docker](https://github.com/3mdeb/edk2-docker).
```

Comment:
@miczyg1 GCC5 is just placeholder name, it works with newer gcc: https://osfw.slack.com/archives/CL8NA71JN/p1613077532025300

Reply:
Understood